### ISSUE TITLE
Menu sometimes detached from its parent [Wayland]

### ISSUE THREAD

**Issue by jeanas** [wbsoft/frescobaldi#1462]
On Fedora, sometimes, when browsing menus, a submenu is detached from its parent menu and appears on a random location on the screen.

![image](https://user-images.githubusercontent.com/37271310/197426341-c65dfdb9-aefc-40aa-a4ac-61755e863377.png)

When running Frescobaldi through the command line, it prints this warning on the console:

```
qt.qpa.wayland: setGrabPopup called with a parent, QtWaylandClient::QWaylandXdgSurface(0x55c199d56540) which does not match the current topmost grabbing popup, QtWaylandClient::QWaylandXdgSurface(0x55c19973e190) According to the xdg-shell protocol, this is not allowed. The wayland QPA plugin is currently handling it by setting the parent to the topmost grabbing popup. Note, however, that this may cause positioning errors and popups closing unxpectedly because xdg-shell mandate that child popups close before parents
```

**Comment by fedelibre** [comment]
This is just another symptom of some Frescobaldi configuration which doesn't play well with Wayland. There are other issues related to Frescobaldi windows and panels broken under Wayland. The workaround is disabling Wayland for Frescobaldi:

    QT_QPA_PLATFORM=xcb frescobaldi

Two years ago I tried to solve what I thought it might be the cause (but I'm not sure), the application ID. See #1258 

Unfortunately my patch created another problem and I was not able to fix it, given my poor programming skills.

**Comment by fedelibre** [comment]
Look at the [issue containing the word wayland in the title](https://github.com/frescobaldi/frescobaldi/issues?q=is%3Aissue+is%3Aopen+wayland+in%3Atitle)

**Comment by fedelibre** [comment]
I've now rebased #1258 and tested it inside a toolbox container.
First of all, by default Wayland is disabled:

```
⬢[fede@toolbox frescobaldi]$ python3 frescobaldi
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
```

If I force Wayland, then I can reproduce the problems described in this issue and the following messages appear:

```
⬢[fede@toolbox frescobaldi]$ QT_QPA_PLATFORM=wayland python3 frescobaldi
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
qt.qpa.wayland: Wayland does not support QWindow::requestActivate()
qt.qpa.wayland: setGrabPopup called with a parent, QtWaylandClient::QWaylandXdgSurface(0x559a63e8acd0) which does not match the current topmost grabbing popup, QtWaylandClient::QWaylandXdgSurface(0x559a63f03630) According to the xdg-shell protocol, this is not allowed. The wayland QPA plugin is currently handling it by setting the parent to the topmost grabbing popup. Note, however, that this may cause positioning errors and popups closing unxpectedly because xdg-shell mandate that child popups close before parents
```

**Comment by fedelibre** [comment]
It seems this Qt bug, which claims to be fixed in 5.15.1:
https://bugreports.qt.io/browse/QTBUG-81952

Am I wrong?
